lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1289,12 +1293,49 @@ CompilerType TypeSystemSwift::GetInstanceType(CompilerType compiler_type) {
   return {};
 }
 
+TypeSystemSwiftTypeRef::TypeSystemSwiftTypeRef(Module &module) {
+  m_module = &module;
+  {
+    llvm::raw_string_ostream ss(m_description);
+    ss << "TypeSystemSwiftTypeRef(\"";
+    module.GetDescription(ss, eDescriptionLevelBrief);
+    ss << "\")";
+  }
+  LLDB_LOGF(GetLogIfAllCategoriesSet(LIBLLDB_LOG_TYPES),
+            "%s::TypeSystemSwiftTypeRef()", m_description.c_str());
+}
+
 TypeSystemSwiftTypeRef::TypeSystemSwiftTypeRef(
-    SwiftASTContext *swift_ast_context)
-    : m_swift_ast_context(swift_ast_context) {
-  m_description = "TypeSystemSwiftTypeRef";
+    SwiftASTContextForExpressions &swift_ast_context)
+    : m_swift_ast_context(&swift_ast_context) {
+  m_description = "TypeSystemSwiftTypeRef(ForExpressions)";
+  LLDB_LOGF(GetLogIfAllCategoriesSet(LIBLLDB_LOG_TYPES),
+            "%sTypeSystemSwiftTypeRef()", m_description.c_str());
 }
 
+SwiftASTContext *TypeSystemSwiftTypeRef::GetSwiftASTContext() const {
```

Comment:
Function `TypeSystemSwiftTypeRef::IsCompatible` is still using `m_swift_ast_context` directly.

Reply:
It might be worth logging if for whatever reason we fail to get a context.